fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -1369,6 +1395,16 @@ ThreadFuture<Void> MultiVersionDatabase::createSnapshot(const StringRef& uid, co
 	return abortableFuture(f, dbState->dbVar->get().onChange);
 }
 
+ThreadFuture<DatabaseSharedState*> MultiVersionDatabase::createSharedState() {
+	auto f = dbState->db ? dbState->db->createSharedState() : ThreadFuture<DatabaseSharedState*>(Never());
+	return abortableFuture(f, dbState->dbVar->get().onChange);
```

Comment:
I think this might have a race condition in it. Instead of the above, you should grab the value of `dbVar` once and then use both its database and onChange signal thereafter. Accessing these two different variables separately here might make it possible that the database changes in the middle.